compiler/test/dotty/tools/dotc/FromTastyTests.scala

```diff
@@ -37,6 +37,11 @@ class FromTastyTests extends ParallelTesting {
         // MatchError in SymDenotation.sourceModule on a ThisType
         "t3612.scala",
 
+        // Cannot handle transparent annotations?
+        "i3129.scala",
```

Comment:
There is a bug pickling the untyped tree.
It can be minimized to 
```scala
class Foo {
  transparent def foo() = {
    class C
    ()
  }
}
```

and reproduced by enabling the  `pickling` printer in `dotty.tools.dotc.config.Printers`.
It will fail printing the tasty at the point where the inconsitency is located:
```scala
**** pickled info of class Foo
Names:
   0: ASTs
   1: <empty>
   2: Foo
   3: <init>
   4: java
   5: lang
   6: java[Qualified . lang]
   7: Object
   8: java[Qualified . lang][Qualified . Object]
   9: <init>[Signed Signature(List(),java.lang.Object)]
  10: Unit
  11: scala
  12: foo
  13: C
  14: Body
  15: annotation
  16: scala[Qualified . annotation]
  17: internal
  18: scala[Qualified . annotation][Qualified . internal]
  19: SourceFile
  20: scala[Qualified . annotation][Qualified . internal][Qualified . SourceFile]
  21: String
  22: java[Qualified . lang][Qualified . String]
  23: <init>[Signed Signature(List(java.lang.String),scala.annotation.internal.SourceFile)]
  24: tests/pos/i3129.scala
  25: Positions
  26: Comments
  27: ASTs
  28: <empty>
  29: Foo
  30: <init>
  31: java
  32: lang
  33: java[Qualified . lang]
  34: Object
  35: java[Qualified . lang][Qualified . Object]
  36: <init>[Signed Signature(List(),java.lang.Object)]
  37: Unit
  38: scala
  39: foo
  40: C
  41: Body
  42: annotation
  43: scala[Qualified . annotation]
  44: internal
  45: scala[Qualified . annotation][Qualified . internal]
  46: SourceFile
  47: scala[Qualified . annotation][Qualified . internal][Qualified . SourceFile]
  48: String
  49: java[Qualified . lang][Qualified . String]
  50: <init>[Signed Signature(List(java.lang.String),scala.annotation.internal.SourceFile)]
  51: tests/pos/i3129.scala
  52: Positions
  53: Comments

Trees:
start = Addr(0), base = 414, current = Addr(0), end = Addr(98)
98 bytes of AST, base = Addr(0)

     0: PACKAGE(96)
     2:   TERMREFpkg 1 [<empty>]
     4:   TYPEDEF(92) 2 [Foo]
     7:     TEMPLATE(74)
     9:       APPLY(7)
    11:         SELECT 9 [<init>[Signed Signature(List(),java.lang.Object)]]
    13:           NEW
    14:             TYPEREF 7 [Object]
    16:               TERMREFpkg 6 [java[Qualified . lang]]
    18:       DEFDEF(8) 3 [<init>]
    21:         PARAMS(0)
    23:         TYPEREF 10 [Unit]
    25:           TERMREFpkg 11 [scala]
    27:         STABLE
    28:       DEFDEF(53) 12 [foo]
    31:         PARAMS(0)
    33:         SHAREDtype 23
    35:         BLOCK(21)
    37:           UNITconst
    38:           TYPEDEF(18) 13 [C]
    41:             TEMPLATE(15)
    43:               APPLY(5)
    45:                 SELECT 9 [<init>[Signed Signature(List(),java.lang.Object)]]
    47:                   NEW
    48:                     SHAREDtype 14
    50:               DEFDEF(6) 3 [<init>]
    53:                 PARAMS(0)
    55:                 SHAREDtype 23
    57:                 STABLE
    58:         TRANSPARENT
    59:         ANNOTATION(22)
    61:           TYPEREF 14 [Body]
    63:             TERMREFpkg 18 [scala[Qualified . annotation][Qualified . internal]]
    65:           UNTYPEDSPLICE(16)
    67:             BLOCK(12)
    69:               EMPTYTREE
    70:               TYPEDEF(9) 13 [C]
    73:                 TEMPLATE(2)
    75:                   TYPEPARAM(3) 6 [java[Qualified . lang]]
    78:                     PACKAGE(6528)
    81:                       SHAREDtype 23
    83:                       ANNOTATION(13)
    85:                         TYPEREF 19 [SourceFile]
    87:                           SHAREDtype 63
    89:                         APPLY(7)
    91:                           SELECT 23 [<init>[Signed Signature(List(java.lang.String),scala.annotation.internal.SourceFile)]]
    93:                             NEW
    94:                               SHAREDtype 85
    96:                           STRINGconst 24 [tests/pos/i3129.scala]
    98:                       ANNOTATEDtpt(51)
   100:                         DEFDEF(64) 33 [java[Qualified . lang]]
   104:                           PARAMS(30)
   106:                             LAMBDA(126)
   108:                               exception occurred while compiling tests/pos/i3129.scala
Exception in thread "main" scala.MatchError: 0 (of class java.lang.Integer)
	at dotty.tools.dotc.core.tasty.TastyFormat$.astTagToString(TastyFormat.scala:515)
	at dotty.tools.dotc.core.tasty.TastyPrinter$TreeSectionUnpickler.printTree$1(TastyPrinter.scala:57)
	at dotty.tools.dotc.core.tasty.TastyPrinter$TreeSectionUnpickler.$anonfun$unpickle$1(TastyPrinter.scala:63)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
	at dotty.tools.dotc.core.tasty.TastyReader.until(TastyReader.scala:125)
	at dotty.tools.dotc.core.tasty.TastyPrinter$TreeSectionUnpickler.printTrees$1(TastyPrinter.scala:63)
	at dotty.tools.dotc.core.tasty.TastyPrinter$TreeSectionUnpickler.printTree$1(TastyPrinter.scala:79)
	at dotty.tools.dotc.core.tasty.TastyPrinter$TreeSectionUnpickler.$anonfun$unpickle$1(TastyPrinter.scala:63)
	at scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:12)
	at dotty.tools.dotc.core.tasty.TastyReader.until(TastyReader.scala:125)
...
```

Reply:
perfect, thanks!